.github/workflows/xrt_master.yml

```diff
@@ -0,0 +1,497 @@
+name: XRT Master
+
+env:  
+  RELEASE: '2024.1'  
+  PIPELINE: 'xrt' 
+  ENV: 'prod'
+
+on:
+  schedule:
+    - cron: '0 17,7 * * *' 
+  
+jobs:
+  build:
+    strategy:      
+      matrix:      
+        include:      
+          - os: centos78      
+            packageType: rpm      
+          - os: centos8      
+            packageType: rpm      
+          - os: rhel9      
+            packageType: rpm      
+          - os: ubuntu2004      
+            packageType: deb      
+          - os: ubuntu2204      
+            packageType: deb      
+        
+    runs-on: [self-hosted, Ubuntu-22.04]
+    steps:  
+      - name: Set env variables    
+        run: |    
+          echo "Setting environment variables..."      
+          echo "XRT_VERSION_PATCH=${GITHUB_RUN_NUMBER}" >> $GITHUB_ENV    
+          echo "PATH=/usr/bin:$PATH" >> $GITHUB_ENV     
+    
+      - name: Checkout PR   
+        uses: actions/checkout@v4 
+        with: 
+          # repository: "${{ github.event.pull_request.head.repo.full_name }}"
+          # ref: "${{ github.event.pull_request.head.ref }}"
+          fetch-depth: 0
+          path: ${{ github.workspace }}/${{ github.run_number }}  
+          submodules: recursive   
+
+      - name: Checkout private repository      
+        uses: actions/checkout@v3   
```

Comment:
I have the feeling it does not make sense to have half of the checkouts at `v4` and the other on obsolete `v3`.